website/pages/blog/creating-a-new-js-plugin.mdx

```diff
@@ -0,0 +1,520 @@
+
+```diff
+export const getTables = async (
++  logger: Logger,
++  spec: Spec
+): Promise<Table[]> => {
+  const table = await getTable();
+  return [table];
+};
+```
+
+### Reading the CSV files
+
+This should be fairly straightforward. We will use the `fs` module to get the list of files in the provided `path` and then load the files using `csv-parse`. We will add a function to get the CSV file paths and another one to parse a file.
+
+Note: Install the `csv-parse` package using `npm install csv-parse`.
+
+```typescript
+//update imports:
+import type { Logger } from "winston";
+import fs from "node:fs/promises";
+import Path from "node:path";
+
+// ...
+
+const getCsvFiles = async (logger: Logger, path: string): Promise<string[]> => {
+  const stats = await fs.stat(path);
+  if (stats.isDirectory()) {
+    const files = await fs.readdir(path, { withFileTypes: true });
+    return files.filter((f) => f.isFile()).map((f) => Path.join(path, f.name));
+  }
+  logger.error("Target path is not a directory.");
+  return [];
+};
+
+const parseCsvFile = async (path: string, csvDelimiter: string): Promise<string[][]> => {
+  const content = await fs.readFile(path);
+  return new Promise<string[][]>((resolve, reject) => {
+    parse(content, { delimiter: csvDelimiter }, (error, records) => {
+      if (error) {
+        reject(error);
+        return;
+      }
+      resolve(records);
+    });
+  });
+}
+```
+
+### Converting the raw CSV data to a table
+
+Our `getTables` function needs to an array of `Table` objects so we will need to convert our raw data. We can use the SDK's `createTable` function which requires us to pass a few options:
+- table name
+- column definitions
+- table resolver - a function that will write the actual records to a provided stream
+
+We will start with the column definitions. First, we need to prepare a column resolver function. [Column resolvers](https://github.com/cloudquery/plugin-sdk-javascript/blob/7a4c4a54d7fcfb266d7cd75c371b86a378ac7969/src/schema/column.ts#L13) are responsible for mapping your data into the columns of the table. In most cases, you will not need to implement anything special, but in some cases, you may need to implement a custom column resolver to fetch additional data or do custom transformations.
+
+In our case, we will just record the data as it comes (this resolver is already in place):
+
+```typescript
+const getColumnResolver = (c: string): ColumnResolver => {
+  return (meta, resource) => {
+    const dataItem = resource.getItem();
+    resource.setColumData(c, (dataItem as Record<string, unknown>)[c]);
+    return Promise.resolve();
+  };
+};
+```
+ 
+We will use the first row of the CSV file to get the column names. We will use the `Utf8` as the default column type for now.
+
+The starter project already contains a function `getTable` that serves as an example so let's modify it to write the data from the CSV file into the stream:
+
+
+```diff
```

Comment:
Using `diff` format doesn't show up that well on the Website. Maybe we can do plain JS/TS code? Then readers can copy paste it too